src/common/outputService/reportOutputService.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfCommand } from '@salesforce/sf-plugins-core';
+import { CliUx } from '@oclif/core';
+import { Flags } from '../base/abstractReportOnPromote';
+import { sObjectToArrayOfKeyValue } from '../utils';
+import { DeploymentResult } from '../types';
+import { tableHeader } from './outputUtils';
+import { AbstractDeploymentResultOutputService, DeploymentResultOutputService } from './deploymentResultOutputService';
+import { OutputFlags } from './outputService';
+
+/* eslint-disable @typescript-eslint/no-empty-interface */
+
+/**
+ * Interface for output methods for report operations.
+ */
+export interface ReportOutputService extends DeploymentResultOutputService {}
+
+/**
+ * Base class.
+ */
+export abstract class AbstractReportOutputService<T extends OutputFlags>
+  extends AbstractDeploymentResultOutputService<T>
+  implements ReportOutputService {}
+
+export class PromoteReportOutputService extends AbstractReportOutputService<OutputFlags> {
+  private operationName: string;
+
+  public constructor(flags: Flags<typeof SfCommand>, deploymentResult: DeploymentResult, operationName: string) {
+    super(
+      {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        verbosity: flags['verbosity'],
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        concise: flags['concise'],
+      },
+      deploymentResult
+    );
+    this.operationName = operationName;
+  }
+
+  public printOpSummary(): void {
+    this.printDeploymentResult();
+  }
+
+  public printDeploymentResult(): void {
+    // convert every field of the deployment result record into a key:value pair.
+    const formattedDeploymentResult = sObjectToArrayOfKeyValue(this.deploymentResult)
+      // rename User.Name field.
+      .map((x) => {
+        x.key = x.key === 'Name' ? 'CreatedByName' : x.key;
+        return x;
+      })
+      // sort by key.
+      .sort((a, b) => (a.key < b.key ? -1 : 1));
+
+    const options = { title: tableHeader(`${this.operationName} Info`) };
+    const columns = { key: { header: 'Key' }, value: { header: 'Value' } };
+
+    CliUx.ux.table(formattedDeploymentResult, columns, options);
```

Comment:
Will we want this logic in the `sf deploy` and `sf resume` commands as well?

Reply:
In those commands we will print a table with the deployed components. But the title and columns will be different.

Reply:
OK, Cool.  My only minor comment then, is do you need `printDeployResults` as a separate method?  Your abstract command is just calling printOpSummary(), which I think is fine.

Reply:
My thinking was that maybe in the future we want report commands to display more info, for example deployed components. So `printOpSummary()` would call `printDeploymentResult() `and `printDeployedComponents()`

Reply:
I'm working on this printDeployedComponents().
It will just be a method calling this CliUx.ux.table with the correct parameters (columns and title defined for deployedcomponents table).
We only need to pass the rows we want to display to this method.